src/scripts/main.js

```diff
@@ -1,3 +1,43 @@
 'use strict';
 
-// write your code here
+const logo = document.getElementsByClassName('logo')[0];
+
+const promise1 = new Promise((resolve, reject) => {
+  const newDiv = document.createElement('div');
+  const newContent = document.createTextNode('Promise was resolved!');
+
+  newDiv.className = 'message';
+  newDiv.appendChild(newContent);
+
+  const body = document.body;
+
+  logo.addEventListener('click', () => {
+    resolve(body.insertAdjacentElement('afterbegin', newDiv));
+  });
+
+  setTimeout(() => {
+    logo.addEventListener('click', () => {
+      reject(body.insertAdjacentElement('afterbegin', newDiv));
+    });
+  }, 3000);
+});
```

Comment:
The promise `promise1` resolves when the logo is clicked, but it also sets up a rejection after 3 seconds. This logic is flawed because the promise can only be resolved or rejected once. Consider restructuring the logic to handle the promise state more effectively.